File: Anul 2/Semestrul 2/MPP/labs/Tema 4-proiect Server-Client Csharp/proiectSCcsharp/angajatPersistence/interfaces/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using proiectSCcsharp.domain;

namespace proiectSCcsharp.repository.interfaces
{
    interface IRepository <ID, T> where T : Entity<ID>
    {
        T add(T entity);
        T remove (ID id);
        T findElement(ID id);
        List<T> getAll();
    }
}

File: Anul 2/Semestrul 2/Ingineria sistemelor soft/labs/Faza 3/proiect/src/main/java/com/example/proiect/repository/IRepository.java
package com.example.proiect.repository;

import com.example.proiect.exceptions.RepoException;

public interface IRepository<T, ID> {
    void add(T entity) throws RepoException;
    void remove(ID id) throws RepoException;
    void update(T entity) throws RepoException;
    T findElement(ID id) throws RepoException;
    Iterable<T> getAll();
}

